### DISCUSSION TITLE
Quick Clarification on Protocol Stack

### DISCUSSION THREAD

**Discussion by Ali78k** [srsran/srsRAN_Project#816] [General Help] [answered]
Hi srsRAN team,

I hope you’re all doing well! I’ve been looking through the documentation and noticed that RRC and MAC procedures are implemented, which is great. However, I’m curious about the status of procedures across the other layers in the 3GPP protocol stack as well as O-RAN Alliance interfaces.
For example, the [gNB interface page](https://docs.srsran.com/projects/project/en/latest/knowledge_base/source/gnb_interfaces/source/index.html#) mentions that E1AP and O1 interfaces are implemented, but I couldn’t find any documentation or confirmation regarding the CU-CP and CU-UP split, or the connection between SMO and gNB.

Could you please clarify:
- Which specific procedures of each layer in the 3GPP protocol stack are implemented in the current release?
- How does the current implementation align with O-RAN Alliance standards? Is there full compliance, or are there areas still in development?

I really appreciate your time, and thanks again for all your hard work!

Best regards,
Ali

**Comment by andrepuschmann** [comment, marked as answer]
Hi Ali - during development we strive for full 3GPP/ORAN spec compliance for all layers/interfaces. There is no document that explains the state of each layer, also this is changing with every release of the software and/or the specs itself. Really the best source is the source. You'll find the E1 interface between CU-CP and UP in the repo. They are not exposed as standalone apps yet but this is something that will done in the near future. O1 is supported as a standalone component outside of the repo that we provide to commercial customers.

**Reply by Ali78k** [reply]
Many thanks @andrepuschmann . I'll take a look at the source code.

**Reply by Ali78k** [reply]
Dear @andrepuschmann , I found some interesting details in the source code about interfaces like F1, E1 and NG, but I'm having trouble understanding the implementation of the 5G NR protocol stack in srsRAN and its compliance with 3GPP standards. Could you provide some guidance?
